_studio/shared/include/mfx_utils.h

```diff
@@ -125,6 +127,18 @@ template<class T> inline T AlignValue(T value, mfxU32 alignment)
     return static_cast<T>((value + alignment - 1) & ~(alignment - 1));
 }
 
+namespace mfx
+{
+// TODO: switch to std::clamp when C++17 support will be enabled
+
+// Clip value v to range [lo, hi]
+template<class T>
+const T clamp(const T v, const T lo, const T hi)
```

Comment:
Why not `const&` ?

Reply:
Suppose it is not good for rvalues and constants: clamp(x+1, 0, 2)

Reply:
Rvalue is not a problem, it will move-construct element which would be used as reference. the main reason is that this code always used for built-in types, so no need to introduce additional indirection with references. If there were some nontrivial classes for which we need ```clamp```, the right solution still will be to use SFINAE to redirect calls for built-in types to implementation with current signature

Reply:
I'd rather use standard signature to avoid surprises when "C++17 support will be enabled"

Reply:
@dmitry-gurulev , agree. Lets follow C++17 prototype.

Reply:
c++17 version is much better for heavy classes with custom comparing operator.
If we are introducing mfx-global utility function to emulate it, lets be consistent and replicate its behavior properly.
And please add this version, as well:
template<class T, class Compare>
constexpr const T& clamp( const T& v, const T& lo, const T& hi, Compare comp );

Reply:
ok, let's do it this way